docs/source/daml/intro/5_Restrictions.rst

```diff
@@ -66,34 +66,38 @@ For example, the simple Iou in :ref:`simple_iou` allowed the no-op where the ``o
   :start-after: -- TRANSFER_TEST_BEGIN
   :end-before: -- TRANSFER_TEST_END
 
-Similarly, you can write a ``Redeem`` choice, which allows the ``owner`` to redeem an ``Iou`` during business hours on weekdays. The choice doesn't do anything other than archiving the ``SimpleIou``. (This assumes that actual cash changes hands off-ledger:)
+Similarly, you can write a ``Redeem`` choice, which allows the ``owner`` to redeem an ``Iou`` *during business hours on weekdays*. Notice the ``Redeem`` choice implementation below confirms that ``getTime`` returns a value that is during business hours on weekdays. If all those checks pass, it doesn't do anything other than archive the ``SimpleIou``. (This assumes that actual cash changes hands off-ledger:)
 
 .. literalinclude:: daml/daml-intro-5/daml/Restrictions.daml
   :language: daml
   :start-after: -- REDEEM_CHOICE_BEGIN
   :end-before: -- REDEEM_CHOICE_END
 
+There's quite a lot going on inside the ``do`` block of the ``Redeem`` choice, with several uses of the ``<-`` operator. ``do`` blocks and ``<-`` deserve a proper explanation which we will get to below. For now, let's look at how the ``Redeem`` choice is exercised in a script.
```

Comment:
There is only one use of the bind operator in ``Redeam``. I would probably avoid drawing attention to it here, as the number of language features we haven't explained that are demonstrated in this code is prohibitive.

The one thing I would consider doing is converting the call to ``assertMsg`` into an if-expression with the case expression in the else clause. I think this is probably easier to read by people coming from non-fp languages than the higher-order function.

Reply:
Fixed with 9993d2672eca9bd445b777a5116bf2c18c548bfa